src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -148,6 +150,11 @@ public ActiveDirectoryDomain(String name, String servers, String site, String bi
 
     @DataBoundConstructor
     public ActiveDirectoryDomain(String name, String servers, String site, String bindName, String bindPassword, TlsConfiguration tlsConfiguration) {
+        // Gives exception if an insecure certificate is used in FIPS mode.
+        if (isFipsNonCompliant(TlsConfiguration.TRUST_ALL_CERTIFICATES.equals(tlsConfiguration))) {
```

Comment:
According to [this](https://github.com/jenkinsci/active-directory-plugin/blob/fa573e714719aeb65b19a4de1c6c09a8079f2f04/src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java#L594-L596) code
```
        private boolean isTrustAllCertificatesEnabled(TlsConfiguration tlsConfiguration) {
                return (tlsConfiguration == null || TlsConfiguration.TRUST_ALL_CERTIFICATES.equals(tlsConfiguration));
        }
```
`null` also means `TRUST_ALL_CERTIFICATES` so that needs to be checked for also, I believe.

Reply:
You could probably just make that method package private and reuse it here and elsewhere.

Reply:
`tlsConfiguration` [has now the default value](https://github.com/jenkinsci/active-directory-plugin/pull/201/files#r1782288943) "JDK_TRUSTSTORE" and this is the value passed, I don't get any `null` value

Reply:
Plenty of ways to make a value `null`; It could be empty on disk (as we noticed when writing that test) it could be set to empty in casc, it could be empty with a POST to /config.xml etc. (the latter probably doesn't work on configureSecurity though) but I hope you get my point :)